### PR TITLE
test(agent): fix service banner and metadata intervals

### PR DESCRIPTION
**Pull request by mafredri**

### CODE REVIEW

File: agent/agent_test.go [hunk 1]
```diff
@@ -1143,7 +1147,9 @@ func TestAgent_Metadata(t *testing.T) {
 					Script:   echoHello,
 				},
 			},
-		}, 0)
+		}, 0, func(client *agenttest.Client, opts *agent.Options) {
+			opts.ReportMetadataInterval = 100 * time.Millisecond
```
Comment (johnstcn) [inline agent/agent_test.go:1151]:
```suggestion
			opts.ReportMetadataInterval = intervalUnit
```

Comment (mafredri) [reply agent/agent_test.go:1151]: I don’t think that’s defined here?

**Review by johnstcn — COMMENTED** [review]
I feel like these tests would be less racy if we could pass a `chan time.Time` between the agent being tested and the unit test itself.

File: agent/agent_test.go (hunk 1 above)
Comment (johnstcn) [reply agent/agent_test.go:1151]: Oops, correct.

**Review by johnstcn — APPROVED** [review]

**Pull request merged**